pyopae/opae.cpp

```diff
@@ -170,16 +169,17 @@ PYBIND11_MODULE(_opae, m) {
       .def("iova", &shared_buffer::iova, shared_buffer_doc_iova())
       .def("fill", &shared_buffer::fill, shared_buffer_doc_fill())
       .def("compare", &shared_buffer::compare, shared_buffer_doc_compare())
-      .def("memoryview", shared_buffer_to_memoryview,
-           shared_buffer_doc_to_memoryview())
       .def_buffer([](shared_buffer &b) -> py::buffer_info {
-    return py::buffer_info(const_cast<uint8_t *>(b.c_type()), b.size());
-      });
+        return py::buffer_info(
+            const_cast<uint8_t *>(b.c_type()), sizeof(uint8_t),
+            py::format_descriptor<uint8_t>::format(), b.size());
+      })
+      .def("__getitem__", shared_buffer_getitem, shared_buffer_doc_getitem())
+      .def("__getitem__", shared_buffer_getslice, shared_buffer_doc_getslice());
```

Comment:
Should both of these be "__getitem__"?

Reply:
Yes. This is "overloaded" to support slicing.
The first method allows using the buffer like this:
```Python
# get byte at offset 256
value = buf[256]
```
The second method allows slicing:
```Python
# get the first 128 bytes
values = buf[0:128]
```